src/test/gather_node_test.cpp

```diff
@@ -262,3 +263,95 @@ TEST_F(GatherNodeTest, FullFlowGatherInNonExitNode) {
     std::copy(nodeRawResults2.begin(), nodeRawResults2.end(), resultTensorData.begin() + nodeRawResults1.size());
     EXPECT_EQ(memcmp((char*)((const void*)gatheredTensor.data()), resultTensorData.data(), resultTensorData.size() * sizeof(float)), 0);
 }
+
+class GatherExitNodeInputHandlerTest : public ::testing::Test {
+protected:
+    char* buffer = nullptr;
+    const std::string tensorName = "example_tensor_name";
+    size_t requestedBufferSize = 20;
+};
+
+class TFSGatherExitNodeInputHandlerTest : public GatherExitNodeInputHandlerTest {
+protected:
+    tensorflow::serving::PredictResponse response;
+};
+
+TEST_F(TFSGatherExitNodeInputHandlerTest, IsBufferSet) {
+    ASSERT_EQ(prepareConsolidatedTensorImpl(&response, buffer, tensorName, requestedBufferSize), StatusCode::OK);
+    EXPECT_NE(buffer, nullptr);
+}
+
+TEST_F(TFSGatherExitNodeInputHandlerTest, BufferPointsToDataInProto) {
+    ASSERT_EQ(prepareConsolidatedTensorImpl(&response, buffer, tensorName, requestedBufferSize), StatusCode::OK);
+    auto it = response.mutable_outputs()->find(tensorName);
+    ASSERT_NE(it, response.mutable_outputs()->end());
+    auto& proto = it->second;
+    EXPECT_EQ(proto.mutable_tensor_content()->data(), buffer);
+}
+
+TEST_F(TFSGatherExitNodeInputHandlerTest, BufferHasCorrectSize) {
+    ASSERT_EQ(prepareConsolidatedTensorImpl(&response, buffer, tensorName, requestedBufferSize), StatusCode::OK);
+    auto it = response.mutable_outputs()->find(tensorName);
+    ASSERT_NE(it, response.mutable_outputs()->end());
+    auto& proto = it->second;
+    EXPECT_EQ(proto.mutable_tensor_content()->size(), requestedBufferSize);
+}
+
+TEST_F(TFSGatherExitNodeInputHandlerTest, TensorAlreadyExistsInProto) {
```

Comment:
We miss similar test for KFS - we should define the behavior in that case as well

Reply:
Added unit test for KFS